nifi-nar-bundles/nifi-standard-services/nifi-distributed-cache-services-bundle/nifi-distributed-cache-server/src/test/java/org/apache/nifi/distributed/cache/server/map/DistributedMapCacheTlsTest.java

```diff
@@ -0,0 +1,142 @@
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        final String port = DistributedMapCacheServer.PORT.getDefaultValue();
+        runner = TestRunners.newTestRunner(Mockito.mock(Processor.class));
+        sslContextService = createSslContextService();
+        runner.addControllerService(sslContextService.getIdentifier(), sslContextService);
+        runner.enableControllerService(sslContextService);
+
+        server = new DistributedMapCacheServer();
+        runner.addControllerService(server.getClass().getName(), server);
+        runner.setProperty(server, DistributedMapCacheServer.PORT, port);
+        runner.setProperty(server, DistributedMapCacheServer.SSL_CONTEXT_SERVICE, sslContextService.getIdentifier());
+        runner.enableControllerService(server);
+
+        client = new DistributedMapCacheClientService();
+        runner.addControllerService(client.getClass().getName(), client);
+        runner.setProperty(client, DistributedMapCacheClientService.HOSTNAME, "localhost");
+        runner.setProperty(client, DistributedMapCacheClientService.PORT, port);
+        runner.setProperty(client, DistributedMapCacheClientService.SSL_CONTEXT_SERVICE, sslContextService.getIdentifier());
+        runner.enableControllerService(client);
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        runner.disableControllerService(client);
+        runner.removeControllerService(client);
+
+        runner.disableControllerService(server);
+        runner.removeControllerService(server);
+
+        runner.disableControllerService(sslContextService);
+        runner.removeControllerService(sslContextService);
+    }
+
+    @Test
+    public void testMapPut() throws IOException {
+        final String key = "keyPut";
+        final String value = "valuePut";
+        assertFalse(client.containsKey(key, serializer));
+        client.put(key, value, serializer, serializer);
+        assertTrue(client.containsKey(key, serializer));
+        assertEquals(value, client.get(key, serializer, deserializer));
+        assertTrue(client.remove(key, serializer));
+        assertFalse(client.containsKey(key, serializer));
+    }
+
+    /**
+     * Create a fresh {@link SSLContext} in order to test mutual TLS authentication aspect of the
+     * distributed cache protocol.
+     *
+     * @return a NiFi {@link SSLContextService}, to be used to secure the distributed cache comms
+     * @throws GeneralSecurityException on SSLContext generation failure
+     * @throws IOException on failure to serialize the SSLContext
+     */
+    private static SSLContextService createSslContextService() throws GeneralSecurityException, IOException {
+        final TlsConfiguration tlsConfiguration = KeyStoreUtils.createTlsConfigAndNewKeystoreTruststore();
+        new File(tlsConfiguration.getKeystorePath()).deleteOnExit();
+        new File(tlsConfiguration.getTruststorePath()).deleteOnExit();
```

Comment:
After rebasing on the current main branch, this will need to change to use `new TemporaryKeyStoreBuilder().build()`:
```suggestion
        final TlsConfiguration tlsConfiguration = new TemporaryKeyStoreBuilder().build();
```